### PR TITLE
WIP: VTOL weathervane

### PR DESCRIPTION
**Pull request by sanderux**
Fixes https://github.com/PX4/Firmware/issues/3679
cherry-picked all commits onto new branch

To enable weathervane mode set param VT_OPT_WV_LND to 1
The VT_OPT_WV_RATE_D param defines the yaw dampening during landing

todo:

- [x] Rename param to vtol_wv_yawrate_scale
- [x] Disable MC_YAWRATE_I when weathervaning
- [x] Add option to weathervane in loiter
- [ ] Implement weathervane strategy when yaw is over-demanded during MC auto flight
- [ ] flight test

### CODE REVIEW

File: src/modules/mc_att_control/mc_att_control_main.cpp [hunk 1]
```diff
@@ -724,6 +732,12 @@ MulticopterAttitudeControl::control_attitude(float dt)
 		}
 	}
 
+	/* weather-vane mode, dampen yaw rate */
+	if (_v_att_sp.disable_mc_yaw_control == true && _v_control_mode.flag_control_velocity_enabled && !_v_control_mode.flag_control_manual_enabled) {
+		float wv_max_yaw_rate = _params.auto_rate_max(2) * _params.vtol_wv_yaw_rate_d;
```
Comment (tumbili) [inline src/modules/mc_att_control/mc_att_control_main.cpp]: @sanderux you can zero the third component of the rates integral here.

File: src/modules/mc_att_control/mc_att_control_main.cpp [hunk 2]
```diff
@@ -201,6 +201,7 @@ class MulticopterAttitudeControl
 		param_t roll_tc;
 		param_t pitch_tc;
 		param_t vtol_opt_recovery_enabled;
+		param_t vtol_wv_yaw_rate_d;
```
Comment (tumbili) [inline src/modules/mc_att_control/mc_att_control_main.cpp]: @sanderux how about changing the name to vtol_wv_yawrate__max? It's not really damping anyway.

**Comment by tumbili** [comment]
@sanderux I fixed the naming. I guess you need to flight test this now.

**Comment by sanderux** [comment]
Thanks, i have a flight window saturday (as it is predicted now).

**Comment by LorenzMeier** [comment]
Looking forward to the testing. I think we should try to de-risk branches in SITL before and turn them into 2-3 test branches so the flight testing becomes tractable.

**Comment by sanderux** [comment]
@tumbili where did your commit go renaming the scale param?

**Comment by LorenzMeier** [comment]
@AndreasAntener @sanderux sounds like we need this now to fix #3728. So should probably focus on this.

File: src/modules/navigator/mission_block.cpp [hunk 3]
```diff
@@ -322,12 +326,18 @@ MissionBlock::mission_item_to_position_setpoint(const struct mission_item_s *ite
 
 	case NAV_CMD_LAND:
 		sp->type = position_setpoint_s::SETPOINT_TYPE_LAND;
+		if(_navigator->get_vstatus()->is_vtol && _param_vtol_wv_land.get()){
```
Comment (LorenzMeier) [inline src/modules/navigator/mission_block.cpp]: We should add this to is_reached() in the mission if the command is a normal waypoint and a 1-2 second timeout occurred.

**Comment by tumbili** [comment]
@sanderux Seems like somebody forced pushed to this branch. I will need to push again from my private laptop.

**Comment by LorenzMeier** [comment]
A force push would be visible in the timeline. Rather looks like they were not pushed at all, or to a different branch.

**Comment by LorenzMeier** [comment]
@tumbili Since the commits are trivial please just quickly re-do them. Don't wait for tonight, as we miss a test window for @AndreasAntener. I want this to move faster now that you're back.

**Comment by tumbili** [comment]
Closing in favour of #3736

**Pull request closed without merging**